handsontable/src/plugins/copyPaste/copyPaste.js

```diff
@@ -750,47 +738,51 @@ export class CopyPaste extends BasePlugin {
       return;
     }
 
-    this.getOrCreateFocusableElement();
-
-    if (this.hot.getSettings().fragmentSelection &&
-        this.focusableElement.getFocusableElement() !== this.hot.rootDocument.activeElement && getSelectionText()) {
+    if (this.hot.getSettings().fragmentSelection) {
       return;
     }
 
     this.setCopyableText();
-    this.focusableElement.focus();
   }
 
   /**
-   * `beforeKeyDown` listener to force focus of focusableElement.
-   *
-   * @private
+   * `document.body` `mouseenter` callback used to work around a Safari's problem with copying/cutting from the
+   * browser's menu.
    */
-  onBeforeKeyDown() {
-    if (!this.hot.isListening() || this.isEditorOpened()) {
-      return;
-    }
-    const activeElement = this.hot.rootDocument.activeElement;
-    const activeEditor = this.hot.getActiveEditor();
+  onSafariMouseEnter() {
+    // If the instance is not listening, the workaround is not needed.
+    if (this.hot.isListening()) {
+      const lastSelectedRange = this.hot.getSelectedRangeLast();
 
-    if (!activeEditor ||
-        (activeElement !== this.focusableElement.getFocusableElement() && activeElement !== activeEditor.select)) {
-      return;
+      if (lastSelectedRange) {
+        const { row: highlightRow, col: highlightColumn } = lastSelectedRange.highlight;
+        const currentlySelectedCell = this.hot.getCell(highlightRow, highlightColumn, true);
+
+        removeContentEditableFromElementAndDeselect(currentlySelectedCell);
+      }
     }
+  }
 
-    this.getOrCreateFocusableElement();
-    this.focusableElement.focus();
+  /**
+   * `document.body` `mouseleave` callback used to work around a Safari's problem with copying/cutting from the
+   * browser's menu.
+   */
+  onSafariMouseLeave() {
+    if (this.hot.isListening()) {
+      const lastSelectedRange = this.hot.getSelectedRangeLast();
+      const { row: highlightRow, col: highlightColumn } = lastSelectedRange.highlight;
+      const currentlySelectedCell = this.hot.getCell(highlightRow, highlightColumn, true);
+
+      if (lastSelectedRange) {
+        makeElementContentEditableAndSelectItsContent(currentlySelectedCell);
+      }
```

Comment:
Shouldn't be that? 🤔 
```suggestion
      if (currentlySelectedCell) {
        makeElementContentEditableAndSelectItsContent(currentlySelectedCell);
      }
```

Reply:
Yup, done in https://github.com/handsontable/handsontable/pull/10342/commits/cd04737fe2df87362f40b2f431943dd173226189